views/marts/product/dim_excludable_servers.view.lkml

```diff
@@ -152,4 +152,23 @@ view: dim_excludable_servers {
     view_label: "Server: Exclusion Reasons"
   }
 
+dimension: all_exclusions {
```

Comment:
Why not edit `has_any_reason`?

Reply:
Good point, I just saw it (doesn't look like we were using it so far). I'll make use of it first then, instead = ).

Reply:
There might be more to it, Looker converts the "No" (don't apply this filter option) to smth. like 
..
WHERE ((dim_server_info."HOSTING_TYPE" ) <> 'Cloud' 
    OR (dim_server_info."HOSTING_TYPE" ) IS NULL) 
    AND (NOT (dim_excludable_servers.has_reason_single_day_security_only or dim_excludable_servers.has_reason_custom_build_version_format or dim_excludable_servers.has_reason_single_day_server_side_telemetry_only or dim_excludable_servers.has_reason_single_day_telemetry_only or dim_excludable_servers.has_reason_ran_tests or dim_excludable_servers.has_reason_active_users__registered_users or dim_excludable_servers.has_reason_no_stripe_installation_found or dim_excludable_servers.has_reason_single_day_user_telemetry_only or dim_excludable_servers.has_reason_restricted_ip or dim_excludable_servers.has_reason_invalid_server_id or dim_excludable_servers.has_reason_internal_email or dim_excludable_servers.has_reason_test_server or dim_excludable_servers.has_reason_community) OR (dim_excludable_servers.has_reason_single_day_security_only or dim_excludable_servers.has_reason_custom_build_version_format or dim_excludable_servers.has_reason_single_day_server_side_telemetry_only or dim_excludable_servers.has_reason_single_day_telemetry_only or dim_excludable_servers.has_reason_ran_tests or dim_excludable_servers.has_reason_active_users__registered_users or dim_excludable_servers.has_reason_no_stripe_installation_found or dim_excludable_servers.has_reason_single_day_user_telemetry_only or dim_excludable_servers.has_reason_restricted_ip or dim_excludable_servers.has_reason_invalid_server_id or dim_excludable_servers.has_reason_internal_email or dim_excludable_servers.has_reason_test_server or dim_excludable_servers.has_reason_community) IS NULL)

which at first sight makes sense (A ^ ~E1 ^ ~E2 ^ ~E3 etc.)but generates fewer results that the "Yes" option - maybe the "IS NULL" at the end(?) throws things off?

Reply:
Thinking out loud here. `has_any_reason` should be always true for any row in this dimension, since it contains rows for servers having at least one reason.


`all_exclusions` seems to actually mean `there's no exclusion reason`. Perhaps the best option would be to find a better name (i.e. `has_no_exclusion_reason`) and keep both?

Reply:
I think I have it reversed:

'has_any_reason' is true (the "Yes" filter path) means that there is at least one reason to exclude, so to get the desired result, in the final filtered query it should be negated (where NOT has_any_reason - show the result with the entries having at least one reason to get excluded - actually excluded).  In case the 'has_any_reason' is false (the "No" filter path) - meaning that there is no reason to exclude - the final query should also negate (where NOT has_any reason...) - to allow for the entries that have no reason to be excluded to be surfaced in the final result. 

"all exclusion reasons" - mean that we try them all and filter out the entries that have at least one exclusion reason (so an extension of company_scorecard_exclusions - the code of which doesn't seem right  - all values are boolean(TRUE/FALSE), why compare with "Yes",NULL?)